### PR TITLE
MULE-8470 - StaxSource NPE - Location can be null

### PR DESCRIPTION
**Pull request by basmussen**
JIRA issue MULE-8470

### CODE REVIEW

File: modules/xml/src/main/java/org/mule/module/xml/stax/StaxSource.java [hunk 1]
```diff
@@ -207,9 +207,17 @@ public void parse() throws SAXException
             }
             catch (XMLStreamException e)
             {
+                // determine location
+                int lineNumber = -1;
+                int columnNumber = -1;
+                if (e.getLocation()!=null)
```
Comment (marianogonzalez) [inline modules/xml/src/main/java/org/mule/module/xml/stax/StaxSource.java]: code format: add space between terms and operators

**Comment by marianogonzalez** [comment]
Hello Ben,

Thanks for reporting this issue and contributing a fix. This PR looks pretty much in order (except for a very very minor code formatting issue), but I was wondering if you tried to do a test case for it as well. I'm thinking that a unit tests which simply instantiates a StaxSource with a mock reader which throws a location less exception could be really helpful.

Have you tried something like that?

Thanks

**Comment by vromero** [comment]
Hey @basmussen,

Any news on this? We are really looking forward to merge this if you can fix the minor aesthetic issue and  create a test case if it's possible.

**Comment by vromero** [comment]
Also, please read and accept the [contributor's agreement](https://api-notebook.anypoint.mulesoft.com/notebooks#bc1cf75a0284268407e4).

**Comment by marianogonzalez** [comment]
I manually merged this PR adding the proper test case.
Thank you for your contribution @basmussen

**Pull request closed without merging**